diff -Naur stfl-0.21.orig/Makefile stfl-0.21/Makefile
--- stfl-0.21.orig/Makefile	2009-06-19 14:30:52.000000000 -0300
+++ stfl-0.21/Makefile	2009-07-20 13:13:43.000000000 -0300
@@ -22,7 +22,7 @@
 
 export CC = gcc -pthread
 export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -fPIC
-export LDLIBS += -lncursesw
+export LDLIBS += -lncursesw -liconv
 
 SONAME  := libstfl.so.0
 VERSION := 0.21
diff -Naur stfl-0.21.orig/python/Makefile.snippet stfl-0.21/python/Makefile.snippet
--- stfl-0.21.orig/python/Makefile.snippet	2009-06-18 05:27:16.000000000 -0300
+++ stfl-0.21/python/Makefile.snippet	2009-07-20 13:14:02.000000000 -0300
@@ -27,7 +27,7 @@
 
 python/_stfl.so python/stfl.py python/stfl.pyc: libstfl.a stfl.h python/stfl.i swig/*.i
 	cd python && swig -python -threads stfl.i
-	gcc -shared -pthread -fPIC python/stfl_wrap.c -I/usr/include/python$(PYTHON_VERSION) \
+	gcc -shared -pthread -fPIC -liconv python/stfl_wrap.c -I/usr/include/python$(PYTHON_VERSION) \
 		-I. libstfl.a -lncursesw -o python/_stfl.so
 	cd python && python -c 'import stfl'
 
