diff -Naur soundtracker-0.6.8.orig/app/Makefile.in soundtracker-0.6.8/app/Makefile.in
--- soundtracker-0.6.8.orig/app/Makefile.in	2006-02-25 10:37:15.000000000 -0300
+++ soundtracker-0.6.8/app/Makefile.in	2006-10-01 17:15:43.000000000 -0300
@@ -716,19 +716,19 @@
 install-exec-local:
 	case `uname` in \
 	  OpenBSD) \
-	    chown root.wheel $(bindir)/soundtracker; \
+	    chown root.wheel $(DESTDIR)$(bindir)/soundtracker; \
 	    ;; \
 	  NetBSD) \
-	    chown root.wheel $(bindir)/soundtracker; \
+	    chown root.wheel $(DESTDIR)$(bindir)/soundtracker; \
 	    ;; \
 	  Darwin) \
-	    chown root:admin $(bindir)/soundtracker; \
+	    chown root:admin $(DESTDIR)$(bindir)/soundtracker; \
 	    ;; \
 	  *) \
-	    chown root:root $(bindir)/soundtracker; \
+	    chown root:root $(DESTDIR)$(bindir)/soundtracker; \
 	    ;; \
 	esac
-	chmod +s $(bindir)/soundtracker
+	chmod +s $(DESTDIR)$(bindir)/soundtracker
 	@echo ""
 	@echo "***"
 	@echo "*** Installing SoundTracker suid root."
