Index: quicktime/ffmpeg/avconfig.h
===================================================================
--- quicktime/ffmpeg/avconfig.h	(revision 946)
+++ quicktime/ffmpeg/avconfig.h	(working copy)
@@ -1,7 +1,7 @@
 #define FFMPEG_CONFIGURATION " "
-#define ARCH_X86 1
-#define TUNECPU generic
-#define __CPU__ 586
+#define ARCH_POWERPC 1 
+#define TUNECPU ppc750
+#define __CPU__ powerpc
 #define HAVE_LOCALTIME_R 1
 #define HAVE_LRINTF 1
 #define HAVE_VHOOK 1
